pkg/build/generator/generator.go

```diff
@@ -735,3 +736,53 @@ func getImageChangeTriggerForRef(bc *buildapi.BuildConfig, ref *kapi.ObjectRefer
 	}
 	return nil
 }
+
+func generateBuildTriggerInfo(requestBuildCauses []*buildapi.BuildTriggerCauses) (buildReasons []*buildapi.BuildTriggerInfo) {
+
+	for _, reason := range requestBuildCauses {
+		switch reason.TriggerType {
+		// the default behavior for no triggertype is a manual build triggered by
+		// the user whenever the user does start-build.
+		// NOTE: This needs either a new const TriggerType that covers jenkins or
+		// another way to differentiate  between builds triggered by a user
+		// clicking a button in the webui, doing oc new-app/new-build/start-build
+		// and jenkins triggering a new build
+		default:
+			buildReasons = append(buildReasons,
+				&buildapi.BuildTriggerInfo{
+					Details: "Manually triggered by user",
+				},
+			)
+		case buildapi.GitHubWebHookBuildTriggerType, buildapi.GenericWebHookBuildTriggerType:
+			buildReasons = append(buildReasons,
```

Comment:
instead of appending here, can you just make a new "reason" (BuildReason sounds better than BuildTriggerInfo) here and then append it after switch?